test/module/shared_model/backend_proto/proto_query_response_factory_test.cpp

```diff
@@ -72,17 +75,24 @@ TEST_F(ProtoQueryResponseFactoryTest, CreateAccountAssetResponse) {
     if (not asset) {
       FAIL() << "could not create common object via factory";
     }
-    assets.push_back(asset);
+    assets.push_back(std::move(asset));
   }
-  auto response = response_factory->createAccountAssetResponse(assets);
+  auto query_response =
+      response_factory->createAccountAssetResponse(assets, kQueryHash);
 
-  ASSERT_TRUE(response);
-  ASSERT_EQ(response->accountAssets().front().accountId(), kAccountId);
-  ASSERT_EQ(response->accountAssets().front().assetId(), kAssetId);
-  for (auto i = 1; i < kAccountAssetsNumber; i++) {
-    ASSERT_EQ(response->accountAssets()[i - 1].balance(),
-              assets[i - 1]->balance());
-  }
+  ASSERT_TRUE(query_response);
+  ASSERT_EQ(query_response->queryHash(), kQueryHash);
+  ASSERT_NO_THROW({
```

Comment:
It is really hard to determine with such block where is the issue. If exception is triggered, than the line of that won't be noticed. It may leave "as is", though would be nice if you come up with the better idea of such situation

Reply:
I will add `SCOPED_TRACE`-s, this will help to at least understand, if `SpecifiedVisitor` worked correctly